cdap-app-fabric/src/main/java/io/cdap/cdap/internal/tethering/TetheringAgentService.java

```diff
@@ -117,8 +119,15 @@ protected long runTask() {
           uri = uri + "?messageId=" + URLEncoder.encode(lastMessageId, "UTF-8");
         }
         HttpResponse resp = TetheringUtils.sendHttpRequest(remoteAuthenticator, HttpMethod.GET,
-                                                           new URI(peer.getEndpoint())
-          .resolve(uri));
+                                                           new URI(peer.getEndpoint()).resolve(uri));
+        /*
+         ** Not sure how to obtain messages here but if we have a way of doing so, then we can call
```

Comment:
To further elaborate on these comments, I was thinking if we can store these messages in a `this.messagesToProcess` property in this class, then we can iterate over `this.messagesToProcess` and extract `peer` specific messages and send to `processNotification` method.